src/pages/CogPage.vue

```diff
@@ -85,9 +96,33 @@ export default class CogPage extends Vue {
           return;
         }
       }
+
       const resp = await fetch(`${c.COGS}/${params.user}/${params.repo}/${params.cog}`);
       const json = await resp.json();
       this.cog = json.results;
+
+      // finding related cogs based on tags
+      const relres = this.cog.tags.map(async (tag) => {
+        const searchResp = await fetch(`${c.SEARCH}/${tag}`);
+        const searchJson = await searchResp.json();
+        return searchJson.results.list;
+      });
+      this.relatedCogs = await Promise.all(relres).then((completed) => {
+        let merged = [];
+        completed.forEach((arr) => {
+          merged = merged.concat(arr);
+        });
+        return merged;
+      });
+      this.relatedCogs = uniqBy(this.relatedCogs, '_id').slice(0, 3);
+
+      if (this.relatedCogs.length < 3) {
+        if (this.cogs.length === 0) {
```

Comment:
so.. you already have the cogs.length check on top. I think we better have it all nice and kept together, because this whole block is a bit confusing.